### PR TITLE
Update Amazon S3 region to endpoint mapping

### PR DESCRIPTION
**Pull request by mlibbey**
https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region has 
additional mappings

### CODE REVIEW

File: plugins/s3_auth/aws_auth_v4.cc [hunk 1]
```diff
@@ -432,27 +435,40 @@ createDefaultRegionMap()
...
+  m["s3.eu-west-1.amazonaws.com"]           = "eu-central-1";
   m["s3-eu-west-1.amazonaws.com"]           = "eu-central-1";
   m["s3.dualstack.eu-west-1.amazonaws.com"] = "eu-central-1";
   /* eu-west-2 */
   m["s3.eu-west-2.amazonaws.com"]           = "eu-west-2";
   m["s3-eu-west-2.amazonaws.com"]           = "eu-west-2";
   m["s3.dualstack.eu-west-2.amazonaws.com"] = "eu-west-2";
   /* sa-east-1 */
+  m["s3.sa-east-1.amazonaws.com"]           = "sa-east-1";
   m["s3-sa-east-1.amazonaws.com"]           = "sa-east-1";
```
Comment (randall) [inline plugins/s3_auth/aws_auth_v4.cc]: Duplicate entry?

File: plugins/s3_auth/aws_auth_v4.cc [hunk 2]
```diff
@@ -412,12 +412,15 @@ createDefaultRegionMap()
   m["s3.dualstack.us-east-2.amazonaws.com"] = "us-east-2";
   /* "us-east-1" */
   m["s3.amazonaws.com"]                     = "us-east-1";
+  m["s3.us-east-1.amazonaws.com"]           = "us-east-1";
   m["s3-external-1.amazonaws.com"]          = "us-east-1";
   m["s3.dualstack.us-east-1.amazonaws.com"] = "us-east-1";
   /* us-west-1 */
+  m["s3.us-west-1.amazonaws.com"]           = "us-west-1"
```
Comment (randall) [inline plugins/s3_auth/aws_auth_v4.cc]: Missing ; at end of line.

Comment (mlibbey) [reply plugins/s3_auth/aws_auth_v4.cc]: thanks!

**Comment by mlibbey** [comment]
Abandoning in favor of #4367

**Comment by zwoop** [comment]
Remember to remove Projects, Milestone etc. :)

**Pull request closed without merging**